cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/CoreSchedulerService.java

```diff
@@ -377,39 +377,37 @@ public void deleteSchedules(Iterable<? extends ScheduleId> scheduleIds) throws N
   @Override
   public void deleteSchedules(ApplicationId appId) {
     checkStarted();
-    List<ScheduleId> deletedSchedules =
-      execute((StoreAndQueueTxRunnable<List<ScheduleId>, RuntimeException>) (store, queue) -> {
-        long deleteTime = System.currentTimeMillis();
-        List<ProgramSchedule> schedules = store.listSchedules(appId);
-        deleteSchedulesInScheduler(schedules);
-        List<ScheduleId> deleted = store.deleteSchedules(appId);
-        for (ScheduleId scheduleId : deleted) {
-          queue.markJobsForDeletion(scheduleId, deleteTime);
-        }
-        for (ProgramSchedule programSchedule : schedules) {
-          adminEventPublisher.publishScheduleDeletion(programSchedule.getScheduleId(), programSchedule);
-        }
-        return deleted;
-      }, RuntimeException.class);
+    execute((StoreAndQueueTxRunnable<Void, RuntimeException>) (store, queue) -> {
+      long deleteTime = System.currentTimeMillis();
+      List<ProgramSchedule> schedules = store.listSchedules(appId);
+      deleteSchedulesInScheduler(schedules);
+      List<ScheduleId> deleted = store.deleteSchedules(appId);
```

Comment:
I meant to remove this list, and do the queue.markJobsForDeletion in the schedules loop.

Reply:
Opened this JIRA to track the bug of schedule: https://issues.cask.co/browse/CDAP-13626